src/sentry/api/endpoints/release_threshold.py

```diff
@@ -0,0 +1,66 @@
+from django.http import HttpResponse
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.api.serializers.rest_framework.environment import EnvironmentField
+from sentry.api.serializers.rest_framework.project import ProjectField
+from sentry.models import Project
+from sentry.models.release_threshold.constants import (
+    THRESHOLD_TYPE_STR_TO_INT,
+    TRIGGER_TYPE_STRING_TO_INT,
+)
+from sentry.models.release_threshold.releasethreshold import (
+    ReleaseThreshold,
+    ReleaseThresholdType,
+    TriggerType,
+)
+
+
+class ReleaseThresholdSerializer(serializers.Serializer):
+    threshold_type = serializers.ChoiceField(choices=ReleaseThresholdType.as_str_choices())
+    trigger_type = serializers.ChoiceField(choices=TriggerType.as_str_choices())
+    value = serializers.IntegerField()
+    window_in_seconds = serializers.IntegerField()
+    project = ProjectField()
+    environment = EnvironmentField(required=False, allow_null=True)
+
+    def validate_threshold_type(self, threshold_type):
+        return THRESHOLD_TYPE_STR_TO_INT[threshold_type]
```

Comment:
Is this just to validate whether the passed threshold type is a valid defined type?

If this returns falsey (eg. 0) will this fail?
And do we need to actually have a mapping of the string -> int for this?

Reply:
This actually converts it to an `int` for us so when we use it in the creation endpoint it gets inserted properly. We'll need the mapping for the conversion and I figured this was a good place for it - lmk if you disagree

Reply:
Oh mmm

I guess i meant the `validate_threshold_type` method.
Does this actually modify the serializer response? or just validates it?

I guess i can just look at how its used in the super 😅

Reply:
I see.
So this _does_ transform the form values
Ok - so we probably wawnt to throw a validation error if the key doesn't exist?